demisto_sdk/commands/common/hook_validations/id.py

```diff
@@ -114,6 +125,101 @@ def _is_non_real_command_found(self, script_data):
+        """
+        is_valid_classifier = True
+        integration_classifier = integration_data.get('classifiers')  # there is only 1 classifier per integration
+        if integration_classifier:
+            # setting initially to false, if the classifier is in the id_set, it will be valid
+            is_valid_classifier = False
+            for classifier in self.classifiers_set:
+                checked_classifier_name = list(classifier.keys())[0]
+                if integration_classifier == checked_classifier_name:
+                    is_valid_classifier = True
+                    break
+            if not is_valid_classifier:  # add error message if not valid
+                error_message, error_code = Errors.integration_non_existent_classifier(integration_classifier)
+                self.handle_error(error_message, error_code, file_path="id_set.json")
+
+        is_valid_mapper = True
+        integration_mapper = integration_data.get('mappers', [''])[0]  # there is only 1 mapper per integration
+        if integration_mapper:
+            # setting initially to false, if the mapper is in the id_set, it will be valid
+            is_valid_mapper = False
+            for mapper in self.mappers_set:
+                checked_mapper_name = list(mapper.keys())[0]
+                if integration_mapper == checked_mapper_name:
+                    is_valid_mapper = True
+                    break
+            if not is_valid_mapper:  # add error message if not valid
+                error_message, error_code = Errors.integration_non_existent_mapper(integration_mapper)
+                self.handle_error(error_message, error_code, file_path="id_set.json")
+
+        return is_valid_classifier and is_valid_mapper
+
+    def _is_classifier_incident_types_found(self, classifier_data):
+        """Check if the classifier incident types were found
+
+        Args:
+            classifier_data (dict): Dictionary that holds the extracted details from the given classfier.
+
+        Returns:
+            bool. Whether the classifier related incident types are found.
+        """
+        is_valid = True
+        classifier_incident_types = set(classifier_data.get('incident_types', set()))
+        if classifier_incident_types:
+            # setting initially to false, if the incident types is in the id_set, it will be valid
+            is_valid = False
+            for incident_type in self.incident_types_set:
+                incident_type_name = list(incident_type.keys())[0]
+                # remove a related incident types if exists in the id_set
+                if incident_type_name in classifier_incident_types:
+                    classifier_incident_types.remove(incident_type_name)
+
+            if not classifier_incident_types:  # if nothing remains, these incident types were all found
+                is_valid = True
+            if not is_valid:  # add error message if not valid
```

Comment:
following my other comment, we can use an `else` clause for the loop.
if reaches the `else`, register the error since the loop iterated over the entire list.

if switching to `else`, keep the comments. they really add to the readability of the code.